src/integer/mat_z.rs

```diff
@@ -29,6 +29,54 @@ mod vector;
 ///     of the [`Z`](crate::integer::Z) matrix
 ///
 /// # Examples
+/// Matrix usage
+/// ```
+/// use math::{
+///     integer::{MatZ, Z},
+///     traits::{GetEntry, SetEntry},
+/// };
+/// use std::str::FromStr;
+///
+/// // instantiate new matrix
+/// let id_mat = MatZ::from_str("[[1,0],[0,1]]").unwrap();
+///
+/// // clone object, set and get entry
+/// let mut clone = id_mat.clone();
+/// clone.set_entry(0, 0, 2);
+/// assert_eq!(clone.get_entry(1, 1).unwrap(), Z::ONE);
+///
+/// // multiplication, transposition and comparison
+/// assert_eq!(id_mat.transpose() * &clone, clone);
+///
+/// // to_string incl. (de-)serialization
+/// assert_eq!("[[1, 0],[0, 1]]", &id_mat.to_string());
+/// assert_eq!(
+///     "{\"matrix\":\"[[1, 0],[0, 1]]\"}",
+///     serde_json::to_string(&id_mat).unwrap()
+/// );
+/// ```
+///
+/// Vector usage
```

Comment:
```suggestion
/// ## Matrix usage
/// ```
/// use math::{
///     integer::{MatZ, Z},
///     traits::{GetEntry, SetEntry},
/// };
/// use std::str::FromStr;
///
/// // instantiate new matrix
/// let id_mat = MatZ::from_str("[[1,0],[0,1]]").unwrap();
///
/// // clone object, set and get entry
/// let mut clone = id_mat.clone();
/// clone.set_entry(0, 0, 2);
/// assert_eq!(clone.get_entry(1, 1).unwrap(), Z::ONE);
///
/// // multiplication, transposition and comparison
/// assert_eq!(id_mat.transpose() * &clone, clone);
///
/// // to_string incl. (de-)serialization
/// assert_eq!("[[1, 0],[0, 1]]", &id_mat.to_string());
/// assert_eq!(
///     "{\"matrix\":\"[[1, 0],[0, 1]]\"}",
///     serde_json::to_string(&id_mat).unwrap()
/// );
/// ```
///
/// ## Vector usage
```

Reply:
Done